services/web/src/components/search/Context.js

```diff
@@ -1,2 +1,14 @@
 import React from 'react';
-export default React.createContext();
+const Context = React.createContext();
+
+export default Context;
+
+export function withSearchProvider(Component) {
```

Comment:
I don't quite get this guy given you seem to only need to use `Search.Provider` to render the provider

Reply:
This meant as hoc for the components, if you got multiple contexts, and need to wrap things in HOC.

Reply:
I don't think it's necessary the provider itself should take care of nested contexts

Reply:
not understanding? this is a way to access the provider, the provider cant take care of it. You have to use either static contentType = Search.Context or withSearchProvider to access the context. 

Anyway I was trying to remove it, to avoid this being hold up by it. 
But then i found https://github.com/bedrockio/bedrock-core/pull/202/files#diff-35364705698e881105cfd24e1d447079b47bb99d25bd65af6f797b55f9d1537fR6

Thinking back this is quite possible why I created this function. Components cant listen to changes from the context, without accessing it via props (thats my understanding) another shot fired at components.

Reply:
Actually they can it's just more convoluted:

```js

getSnapshotBeforeUpdate() {
  return this.context;
}

componentDidUpdate(lastProps, lastState, lastContext) {
   // ...
}
```

Not suggesting this is better (it's not) just that it is possible. We were at one point doing this with `session.js` but I found it to be unnecessary because (more to the point) a component shouldn't really need or want to concern itself with changes the context, only it's own state or props. This is kind of a side discussion anyway though, I don't think it's related here but it's been months so I'm going to have to sit down and have another look here. Will get to it soon I promise 👍

Reply:
ohh see let me give that a try